Riot/Modules/MatrixKit/Models/Room/MXKRoomBubbleCellData.m

```diff
@@ -55,12 +57,14 @@ - (instancetype)initWithEvent:(MXEvent *)event andRoomState:(MXRoomState *)roomS
             senderId = event.sender;
             targetId = [event.type isEqualToString:kMXEventTypeStringRoomMember] ? event.stateKey : nil;
             roomId = roomDataSource.roomId;
-            senderDisplayName = [roomDataSource.eventFormatter senderDisplayNameForEvent:event withRoomState:roomState];
-            senderAvatarUrl = [roomDataSource.eventFormatter senderAvatarUrlForEvent:event withRoomState:roomState];
+
+            MXRoomState *profileRoomState = RiotSettings.shared.roomScreenUseOnlyLatestProfiles ? roomDataSource.roomState : roomState;
```

Comment:
I wonder should we rename `roomState` to `eventRoomState` and either include a comment that `roomDataSource.roomState` is the latest room state or assign it to a `latestRoomState` variable first. Otherwise it makes this line of code a bit obscure for first time readers.

Reply:
Agreed. Especially since it is important that it is actually the roomState from the event on init (to get correct encrypted status), renaming would make it clearer.

Reply:
Actually scratch that, since there are multiple objects using the same interface (with this init, and even more with the addEvent selector), there might be a bunch of places where the renaming doesn't make as much sense (even if it's technically right, I think it's gonna be too wordy for no particular reason). I'll go with a precise comment over this code, I hop it'll be enough to be understandable.